src/server/game/Grids/Notifiers/GridNotifiers.cpp

```diff
@@ -256,7 +256,12 @@ void AIRelocationNotifier::Visit(CreatureMapType &m)
         Creature* c = iter->GetSource();
         CreatureUnitRelocationWorker(c, &i_unit);
         if (isCreature)
-            CreatureUnitRelocationWorker((Creature*)&i_unit, c);
+        {
+            if (i_unit.IsSummon() && !i_unit.ToTempSummon()->IsInitialized())
+                continue;
+            else
+                CreatureUnitRelocationWorker((Creature*)&i_unit, c);
+        }
     }
```

Comment:
This can be simplified to:

```c++
    for (CreatureMapType::iterator iter = m.begin(); iter != m.end(); ++iter)
    {
        Creature* c = iter->GetSource();
        CreatureUnitRelocationWorker(c, &i_unit);
        if (isCreature && (!i_unit.IsSummon() || i_unit.ToTempSummon()->IsInitialized()))
            CreatureUnitRelocationWorker((Creature*)&i_unit, c);
    }
```